projects/plugins/jetpack/extensions/blocks/premium-content/edit.js

```diff
@@ -217,6 +218,45 @@ function Edit( props ) {
 
 	const { isSelected, className } = props;
 
+	const selectedBlock = useSelect( selector => selector( 'core/block-editor' ).getSelectedBlock() );
+
+	useEffect( () => {
+		if ( isSelected ) {
+			return; // If this block is selected then leave the focused tab as it was.
+		}
+
+		if ( ! selectedBlock ) {
+			return; // Sometimes there isn't a block selected, e.g. on page load.
+		}
+
+		// Confirm that the selected block is a descendant of this one.
+		if (
+			! select( 'core/block-editor' ).getBlockParents( selectedBlock.clientId ).includes( clientId )
+		) {
+			return;
+		}
+
+		if (
+			'premium-content/logged-out-view' === selectedBlock.name ||
+			select( 'core/block-editor' ).getBlockParentsByBlockName(
+				selectedBlock.clientId,
+				'premium-content/logged-out-view'
+			).length
+		) {
+			selectTab( tabs[ 1 ] );
+			return;
+		} else if (
```

Comment:
Since we are using `select( 'core/block-editor' )` more than once, we could extract it to a const. 

Right now I've seen that you are using the literal string `premium-content/logged-out-view` to check against the block name, we currently have this string exported as a constant in the logged-out-view/index.js, maybe we could import the constant and use that instead (if the block name changes we would only need to modify one file instead of two).

Also, when selecting the tab we are using a magic number to select the tab. Maybe we could create a map between the view name and the tab number so that we could select a tab like so: `selectTab( tabs[LOGGED_OUT_VIEW] )` or even better `selectTab( LOGGED_OUT_VIEW )`. I'm aware this last suggestion is out of scope, but I think it increases the readability of the code.

```suggestion
                 const blockEditor = select( 'core/blcok-editor' );
		
		// Confirm that the selected block is a descendant of this one.
		if ( ! blockEditor.getBlockParents( selectedBlock.clientId ).includes( clientId ) ) {
			return;
		}

		if ( LOGGED_OUT_VIEW === selectedBlock.name ||
			blockEditor.getBlockParentsByBlockName(
				selectedBlock.clientId,
				LOGGED_OUT_VIEW
			).length
		) {
			selectTab( tabs[ 1 ] );
			return;
		} else if (
```

Reply:
> Right now I've seen that you are using the literal string premium-content/logged-out-view

That's happening in multiple places across this block for logged out and subscriber, we could have a separate clean-up PR for that.

Reply:
> Since we are using `select( 'core/block-editor' )` more than once, we could extract it to a const.

We can also destructure to only get the actual functions we need:
```es6
const { getBlockParents, getBlockParentsByBlockName } = select( 'core/block-editor' );
```

---

I don't have a strong opinion about turning the block name into a constant.
It's handy and we [constant-ize it in PHP already](https://github.com/Automattic/jetpack/blob/fee24dc45c68329fd344493531a4cb16217efda7/projects/plugins/jetpack/extensions/blocks/premium-content/logged-out-view/logged-out-view.php#L13), but Dean has some good arguments against it.
I personally like being able to search strings directly, but really, whatever works for y'all. 😄